src/main/java/org/openstreetmap/atlas/checks/validation/areas/AreasWithHighwayTagCheck.java

```diff
@@ -34,34 +45,77 @@ public AreasWithHighwayTagCheck(final Configuration configuration)
     @Override
     public boolean validCheckForObject(final AtlasObject object)
     {
-        return object instanceof Area;
+        return object instanceof Area || (object instanceof Edge && ((Edge) object).isMasterEdge());
     }
 
     @Override
     protected Optional<CheckFlag> flag(final AtlasObject object)
     {
-        // See configuration for initial check filter; highway=pedestrian & area=yes is a valid
-        // Area/Highway tag combination.
-        final Area area = (Area) object;
-        final Optional<String> highwayTag = area.getTag(HighwayTag.KEY);
-        // First check if highway tag exists
-        if (highwayTag.isPresent())
+        if (!this.isFlagged(object.getOsmIdentifier()))
         {
-            // Send special instructions for pedestrian highways missing area=yes tag
-            final int localizedInstruction = Validators.isOfType(area, HighwayTag.class,
-                    HighwayTag.PEDESTRIAN) && !Validators.isOfType(area, AreaTag.class, AreaTag.YES)
-                            ? 1 : 0;
-
-            return Optional.of(this.createFlag(object,
-                    this.getLocalizedInstruction(localizedInstruction, object.getOsmIdentifier())));
-
+            return object.getTag(HighwayTag.KEY)
+                    .map(tagString -> HighwayTag.valueOf(tagString.toUpperCase()))
+                    // If the tag isn't one of the validHighwayTags, we want to flag it.
+                    .filter(tag -> isUnacceptableAreaHighwayTagCombination(object, tag)).map(tag ->
+                    {
+                        final String instruction;
+                        if (isNotOsmStandard(tag))
```

Comment:
If that is the case it might be worth clarifying this in the instruction. Perhaps something where we give a suggested change like: "The way with OSM ID 1234567 has a highway value of highway=footway, which should not have an area=yes tag. Consider changing the highway tag to highway=pedestrian"

Reply:
I think Bentley's suggestion is fine. The more I think about it, the more I feel it's probably alright in this instance to suggest pedestrian. We run the risk of suggesting an incorrect fix when there should be no highway tag, however, with this check I think the majority of footway tags should probably just be pedestrian anyway.